tests/integration/cartography/graph/test_querybuilder_rel_subsets.py

```diff
@@ -0,0 +1,140 @@
+from cartography.client.core.tx import load_graph_data
+from cartography.graph.querybuilder import build_ingestion_query
+from tests.data.graph.querybuilder.sample_data.partial_relationships import INTERESTING_NODE_WITH_PARTIAL_RELATIONSHIPS
+from tests.data.graph.querybuilder.sample_data.partial_relationships import MERGE_SUB_RESOURCE_QUERY
+from tests.data.graph.querybuilder.sample_data.partial_relationships import MERGE_WORLD_ASSET_QUERY
+from tests.data.graph.querybuilder.sample_models.interesting_asset import InterestingAssetSchema
+
+
+def test_load_graph_data_subset_of_relationships(neo4j_session):
+    """
+    Test load_graph_data() if a schema defines multiple relationships but only a subset of them are present in our data.
+
+    In this test case, the following relationships are possible:
+        (:InterestingAsset)<-[:RELATIONSHIP_LABEL]-(:SubResource)
+        (:InterestingAsset)-[:ASSOCIATED_WITH]->(:HelloAsset)
+        (:InterestingAsset)<-[:CONNECTED]-(:WorldAsset)
+    but our test data does not include a HelloAsset.
+    """
+    # Arrange: add (:SubResource{id:sub-resource-id}) and (:WorldAsset{id: world-asset-id}) to the test graph
+    neo4j_session.run(MERGE_SUB_RESOURCE_QUERY)
+    neo4j_session.run(MERGE_WORLD_ASSET_QUERY)
+
+    # Act
+    query = build_ingestion_query(InterestingAssetSchema())
+    load_graph_data(
+        neo4j_session,
+        query,
+        INTERESTING_NODE_WITH_PARTIAL_RELATIONSHIPS,
+        lastupdated=1,
+        sub_resource_id='sub-resource-id',
+    )
+
+    # Assert that the InterestingAsset to SubResource relationship exists
+    expected = {
+        ('interesting-node-id', 'sub-resource-id'),
+    }
+    result = neo4j_session.run(
+        """
+        MATCH (n1:InterestingAsset)<-[:RELATIONSHIP_LABEL]-(n2:SubResource) RETURN n1.id, n2.id;
+        """,
+    )
+    actual = {
+        (r['n1.id'], r['n2.id']) for r in result
+    }
+    assert actual == expected
+
+    # Assert that the InterestingAsset to HelloAsset relationship does NOT exist
+    expected = {
+        ('interesting-node-id', None),
+    }
+    result = neo4j_session.run(
+        """
+        MATCH (n1:InterestingAsset)
+        OPTIONAL MATCH (n1)<-[:ASSOCIATED_WITH]-(n2:HelloAsset)
```

Comment:
I think you meant to write an outward relationship direction here. Still, we can rewrite it this way:
```suggestion
        OPTIONAL MATCH (n1)--(n2:HelloAsset)
```